examples/api/lib/material/selection_area/selection_area.2_b.dart

```diff
@@ -0,0 +1,369 @@
+                    insideSelection.add(const TextSpan(text: '\n'));
+                  }
+                }
+                // Check if this span contains content after the selection.
+                if (globalNewStartAfterSelection == endOffset && newStartAfterSelection < rawText.length) {
+                  afterSelection.add(
+                    TextSpan(
+                      style: child.style,
+                      text: rawText.substring(newStartAfterSelection),
+                    ),
+                  );
+                }
+              }
+            } else if (count >= endOffset) {
+              // Collect spans after selection.
+              afterSelection.add(TextSpan(style: child.style, text: rawText));
+            } else {
+              // Collect spans inside selection.
+              final int newStart = min(endOffset - count, rawText.length);
+              final int globalNewStart = count + newStart;
+              if (rangeIndex == 0) {
+                insideSelection.add(const TextSpan(text: '\n'));
+                insideSelection.add(const TextSpan(text: '\n'));
+                // Can either wrap with a SelectionArea so the WidgetSpan can have its
+                // own contained selection. Or a SelectionContainer.disabled to completely
+                // disable the selection of the WidgetSpan contents.
+                insideSelection.add(WidgetSpan(child: SelectionArea(child: DetailBox(children: _initSpans(plainText)))));
+                if (ranges.length == 1) {
+                  insideSelection.add(const TextSpan(text: '\n'));
+                  insideSelection.add(const TextSpan(text: '\n'));
+                }
+              }
+              // Check if this span contains content after the selection.
+              if (globalNewStart == endOffset && newStart < rawText.length) {
+                afterSelection.add(TextSpan(style: child.style, text: rawText.substring(newStart)));
+              }
+            }
+            count += rawText.length;
+          }
+        } else if (child is WidgetSpan) {
+          if (count < startOffset) {
+            beforeSelection.add(child);
+          } else if (count >= endOffset) {
+            afterSelection.add(child);
+          } else {
+            insideSelection.add(child);
+          }
+          count += 1;
+        }
+        return true;
+      });
+      dataSourceMap[indexOfText] = TextSpan(
+        style: dataSourceMap[indexOfText]!.style,
+        children: <InlineSpan>[
+          ...beforeSelection,
+          ...insideSelection,
+          ...afterSelection,
+        ],
+      );
+    }
+    // Rebuild column contents.
+    for (final MapEntry<int, TextSpan> entry in dataSourceMap.entries) {
+      newText.add(
+        SelectionListener(
+          onSelectionChanged: (SelectionDetails selectionDetails) {
+            selectionDetailsMap[entry.key] = selectionDetails;
+          },
+          child: Text.rich(
```

Comment:
Mainly because given that any complex widget tree could be under `SelectionArea` we decided it was best to let the user decide if they only want to listen to a portion of thats trees selection changes rather than the entire tree.